tests/vegalite/v5/test_api.py

```diff
@@ -23,6 +23,7 @@
 import polars as pl
 
 import altair as alt
+from altair.utils.schemapi import Undefined
```

Comment:
What's the reasoning behind importing `Undefined` from the submodule?

Reply:
This was a similar change to that of [v5/test_params](https://github.com/vega/altair/pull/3482/files/1e0c12d2e10ddf23636233d1df6f432cccbe861d#diff-5bc52267a53307a825084d94af60cbf84eee6b53fa642e206a469a641802bc03).

While currently `Undefined` is available at the top-level, this may not be the case in a future version (hypothetically).

Using a *more* explicit import isolates these tests to the core functionality they are testing (`v5/api.py`) without imposing the dependencies of high level packages.

---

### Some scenarios
- `Undefined` is removed from top-level
	- This version would not need to be modified
	- Existing version would need updating
		- Then requires trying to follow where the new import location would be
- `Undefined` is removed from `altair.utils.__all__`
	- This version leaves a clear trace of where it was before, hinting where you should look for changes
		- May not even require a change, if it remained in `schemapi.py`, but was exported to `utils.__all__`
	- Existing version may/may not need updating
		- But since we relied on a higher level export, the origin of `Undefined` isn't clear

---

When refactoring code, things like [rename module](https://code.visualstudio.com/docs/python/editing#_rename-module) and [move symbol](https://code.visualstudio.com/docs/python/editing#_move-symbol) will be able to use these absolute imports and keep them in sync - since they are clearly traceable to the origin of the symbol.

Reply:
While I'd consider it unlikely right now that we remove `Undefined` from the top-level as it might break a lot of user code (personally, I'm working with a codebase which contains a lot of `alt.Undefined` references), I also don't see a downside so fine me. Thanks for the thorough explanation!